src/frontend/components/Chat/SharedChatComponents/ChatConversationDisplayer/helpers.tsx

```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+
+import { ChatMessageContent } from 'components/Chat/SharedChatComponents/ChatMessageContent';
+import { ChatMessageItem } from 'components/Chat/SharedChatComponents/ChatMessageItem';
+import { ChatPresenceItem } from 'components/Chat/SharedChatComponents/ChatPresenceItem';
+import { ChatMessageType } from 'data/stores/useMessagesStore';
+import { ChatPresenceType } from 'data/stores/usePresenceItemsStore';
+
+export const computeChatItems = (
+  messages: ChatMessageType[],
+  presences: ChatPresenceType[],
+): JSX.Element[] => {
```

Comment:
Can you try to remove usage of `for ... of ...` and use `map` or `reduce` instead ? Also, you can maybe use `presences.shift()` to get the first value and remove it from the array and avoir using an index counter ?

Reply:
I changed the way it works, to use more array functions. It's now more readable but it may be slower too.
done in commit [3180aa9724c2740bf54a739b636478b6530d0b64](https://github.com/openfun/marsha/pull/1279/commits/3180aa9724c2740bf54a739b636478b6530d0b64)